src/middleware.ts

```diff
@@ -0,0 +1,65 @@
+import type { Request, Response, NextFunction } from 'express';
+import { ErrorHTTP } from './error';
+
+export interface ShowErrorOptions {
+  logger?(): void,
+}
+
+/**
+ * An express.js middleware used for showing errors in your logs
+ * and converting them to JSONP enabled response messages.
+ *
+ * It will log any >=500 status codes using console.error or a custom
+ * logger and update the error message to "Internal Server Error" to 
+ * obfuscate any stack or native-level errors that you don't want 
+ * to expose to users.
+ *
+ * Any status codes <500 are assumed to contain error messages crafted
+ * for public consumption.
+ * @param {Object} options - Options for the middleware
+ * @param {Function} options.logger Custom logger function for logging the errors
+ * defaults to console.error
+ */
+export function showError(options?: ShowErrorOptions) {
+  const logger = options?.logger || console.error;
+  
+  // NOTE: next is needed, even if not used, per https://expressjs.com/en/guide/using-middleware.html
+  return function(err: any, req: Request, res: Response, next: NextFunction) { // eslint-disable-line @typescript-eslint/no-unused-vars
+    err.status = err.status || 500;
+  
+    // Output unexpected errors to console but hide them from public eyes.
+    if (err.status >= 500) {
+      err.url = req.url;
+      err.method = req.method;
+      logger(err);
+      err.message = 'Internal Server Error';
+    }
+  
+    const data: Record<string, any> = { 
+      message: err.message
+    };
+  
+    // For non-500 ErrorHTTP objects send over any additional keys.
+    // This error is one that we crafted ourselves deliberately for
+    // public consumption.
+    if (err instanceof ErrorHTTP && err.status < 500) {
+      for (const k in err) {
+        if (k === 'status') continue;
+        data[k] = (err as any)[k];
```

Comment:
do you need the `any` casting here since you know it's of type `ErrorHTTP` here?

Reply:
Without it, typescript assumes this is an ErrorHTTP instance, which gives me this type error:

```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'ErrorHTTP'.
  No index signature with a parameter of type 'string' was found on type 'ErrorHTTP'.
```

Open to suggestions on how to change this!

Reply:
Ahh yeah that makes sense since you're trying to access the value at `err[k]`. It seems like the only thing we're trying to do here is create a new object called `data` that doesn't have the key `status` in it? You could replace the entire for loop with something like:
```suggestion
        const { status, ...data } = err;
```
This will create a new `data` obj without the status `property`. Downside is that you might get a warning or error for status being unused.

Reply:
Thanks for the suggestion! I included your commit but had to make a couple more changes and ultimately landed on the changes in 0652b1a.